tools/publisher/src/subcommand/generate_version_manifest.rs

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::fs::Fs;
+use crate::package::{discover_package_manifests, read_packages};
+use anyhow::{bail, Context, Result};
+use serde::{Deserialize, Serialize};
+use smithy_rs_tool_common::git::GetLastCommit;
+use smithy_rs_tool_common::package::PackageCategory;
+use smithy_rs_tool_common::shell::{self, ShellOperation};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::path::Path;
+use tracing::info;
+
+pub async fn subcommand_generate_version_manifest(
+    smithy_build_path: &Path,
+    examples_revision: &str,
+    location: &Path,
+) -> Result<()> {
+    verify_crate_hasher_available()?;
+
+    let smithy_rs_revision = GetLastCommit::new(std::env::current_dir()?)
+        .spawn()
+        .await
+        .context("get smithy-rs revision")?;
+    info!("Resolved smithy-rs revision to {}", smithy_rs_revision);
+
+    let smithy_build_root = SmithyBuildRoot::from_file(smithy_build_path)?;
+    let manifests = discover_package_manifests(location.into())
+        .await
+        .context("discover package manifests")?;
+    let packages = read_packages(Fs::Real, manifests)
+        .await
+        .context("read packages")?;
+
+    let mut crates = BTreeMap::new();
+    for package in packages {
+        // Don't include examples
+        if let PackageCategory::Unknown = package.category {
+            continue;
+        }
+
+        let mut model_hash = None;
+        if package.handle.name.starts_with("aws-sdk-") {
```

Comment:
how is this different than `PackageCategory::AwsSdk`?

Reply:
It isn't. Not sure why I did it that way. Will switch to using the category.